src/home_robot/home_robot/mapping/semantic/categorical_2d_semantic_map_module.py

```diff
@@ -590,7 +745,12 @@ def _update_global_map_and_pose_for_env(
         """Update global map and pose and re-center local map and pose for a
         particular environment.
         """
+
         global_map[e, :, lmb[e, 0] : lmb[e, 1], lmb[e, 2] : lmb[e, 3]] = local_map[e]
+        if self.record_instance_ids:
+            global_map = self.add_instance_ids_to_global_map(
```

Comment:
So clustering the map into object instances doesn't happen at every step but instead when we update the global map. This is not how I imagined it but it makes sense